app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java

```diff
@@ -1289,7 +1293,7 @@ public static BgReading bgReadingInsertFromJson(String json, boolean do_notifica
     }
 
     // TODO this method shares some code with above.. merge
-    public static void bgReadingInsertFromInt(int value, long timestamp, boolean do_notification) {
+    public static void bgReadingInsertFromInt(int value, long timestamp, double margin, boolean do_notification) {
```

Comment:
I'm slightly concerned about adding more use of `double` to represent `milliseconds`